test/e2e/aws_test.go

```diff
@@ -431,8 +430,64 @@ var _ = Describe("functional tests", func() {
 			deleteCluster(setup.namespace, setup.clusterName)
 		})
 	})
+
+	Describe("Cluster should be highly available", func() {
+		It("Control plane should be functional after failure of one master node", func() {
+			By("Creating a workload cluster with single control plane")
+			setup.multipleAZ = true
+			clusterK8sClient := makeSingleControlPlaneCluster(setup)
+
+			By("Creating the second Control Plane Machine in second AZ")
+			awsMachineName := setup.cpAWSMachinePrefix + "-1"
+			bootstrapConfigName := setup.cpBootstrapConfigPrefix + "-1"
+			machineName := setup.cpMachinePrefix + "-1"
+			setup.availabilityZone = availabilityZones[1]
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Creating the third Control Plane Machine in third AZ")
+			awsMachineName = setup.cpAWSMachinePrefix + "-2"
+			bootstrapConfigName = setup.cpBootstrapConfigPrefix + "-2"
+			machineName = setup.cpMachinePrefix + "-2"
+			setup.availabilityZone = availabilityZones[2]
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Creating the MachineDeployment in second AZ")
+			setup.availabilityZone = availabilityZones[1]
+			setup.subnetId = nil
+			createMachineDeployment(setup)
+
+			By("Deleting Control Plane Machine in second AZ")
+			deleteMachine(machineName, setup.namespace)
+
+			// Create a kubernetes resource on the workload cluster to
+			// test the ability to write to the API server
+			By("Creating a kubernetes resource")
+			createDummyResource("dummy-pod", metav1.NamespaceDefault, clusterK8sClient)
+
+			By("Deleting the Cluster")
+			deleteCluster(setup.namespace, setup.clusterName)
+		})
+	})
 })
 
+func createDummyResource(name, namespace string, k8sClient crclient.Client) {
```

Comment:
Instead of creating a Pod here, I would probably just create a Namespace to test the functionality of the API server.